web/settings/modulconfiglp.php

```diff
@@ -3578,6 +3693,97 @@ function visibility_kia_soccalclp2() {
+														<input type="radio" name="soc_evcc_type_lp2" id="soc_evcc_type_kia_lp2" value="kia"<?php if($soc_evcc_type_lp2old == 'kia') echo " checked=\"checked\"" ?>>Kia
+													</label>
+													<label class="btn btn-outline-info<?php if($soc_evcc_type_lp2old == 'hyundai') echo " active" ?>">
+														<input type="radio" name="soc_evcc_type_lp2" id="soc_evcc_type_hyundai_lp2" value="hyundai"<?php if($soc_evcc_type_lp2old == 'hyundai') echo " checked=\"checked\"" ?>>Hyundai
+													</label>
+													<label class="btn btn-outline-info<?php if($soc_evcc_type_lp2old == 'nissan') echo " active" ?>">
+														<input type="radio" name="soc_evcc_type_lp2" id="soc_evcc_type_nissan_lp2" value="nissan"<?php if($soc_evcc_type_lp2old == 'nissan') echo " checked=\"checked\"" ?>>Nissan
+													</label>
+													<label class="btn btn-outline-info<?php if($soc_evcc_type_lp2old == 'renault') echo " active" ?>">
+														<input type="radio" name="soc_evcc_type_lp2" id="soc_evcc_type_renault_lp2" value="renault"<?php if($soc_evcc_type_lp2old == 'renault') echo " checked=\"checked\"" ?>>Renault
+													</label>
+												</div>
+											</div>
+											<span class="form-text small">Auswahl Fahrzeugtyp</span>
+										</div>
+									</div>
+									<div class="form-row mb-1">
+										<label for="soc_evcc_username_lp2" class="col-md-4 col-form-label">Benutzername</label>
+										<div class="col">
+											<input class="form-control" type="email" name="soc_evcc_username_lp2" id="soc_evcc_username_lp2" value="<?php echo $soc_evcc_username_lp2old ?>">
+											<span class="form-text small">
+												Email Adresse des Logins
+											</span>
+										</div>
+									</div>
+									<div class="form-row mb-1">
+										<label for="soc_evcc_password_lp2" class="col-md-4 col-form-label">Passwort</label>
+										<div class="col">
+											<input class="form-control" type="password" name="soc_evcc_password_lp2" id="soc_evcc_password_lp2" value="<?php echo $soc_evcc_password_lp2old ?>">
+											<span class="form-text small">
+												Passwort des Logins
+											</span>
+										</div>
+									</div>
+									<div class="form-row mb-1">
+										<label for="soc_evcc_vin_lp2" class="col-md-4 col-form-label">VIN</label>
+										<div class="col">
+											<input class="form-control" type="text" name="soc_evcc_vin_lp2" id="soc_evcc_vin_lp2" value="<?php echo $soc_evcc_vin_lp2old ?>">
+											<span class="form-text small">
+												Vollständige VIN des Fahrzeugs
+											</span>
+										</div>
+									</div>
+									<div class="form-row mb-1">
+										<label for="soc_evcc_token" class="col-md-4 col-form-label">Token</label>
+										<div class="col">
+											<input class="form-control" type="text" name="soc_evcc_token" id="soc_evcc_token" value="<?php echo $soc_evcc_tokenold ?>">
```

Comment:
Das Input Element verwendet die selbe ID wie bei LP1. IDs müssen im HTML jedoch eindeutig sein.
Der Name des Elements ist ebenfalls identisch zu LP1. Das wird bei potentiell zwei aktivierten SoC mit diesem Modul zu Problemen führen. Namen in Formularen müssen (abgesehen von Radio-Buttons) ebenfalls eindeutig sein.

Reply:
Also muss ich den Token auch in der openWB.conf zweimal anlegen lassen, obwohl der Token immer gleich ist für beide LP? Es soll ja explizit der Selbe sein, den man bei LP2 auch nicht erneut eintragen muss wenn in LP1 erfolgt (und vice-versa).